pkg/api/v1/types.go

```diff
@@ -2110,6 +2110,8 @@ type PodSpec struct {
 	// If specified, the pod's scheduling constraints
 	// +optional
 	Affinity *Affinity `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
+	// SchedulingMismatchedPredicateResults Summary of false podFitsOnNode predicates for investigating unscheduled pods.
```

Comment:
- end user facing object, so we won't be nesting more detail in it necessarily.
- pure `map[string]int` reflects exactly what we want for this use case.
i will work on a struct implementation in a branch anyway, but just thought id clarify those details just in case it changes your mind .